File: src/main/java/com/stockpro/stockpro/service/administration/RoleService.java
package com.stockpro.stockpro.service.administration;

import com.stockpro.stockpro.entity.administration.Role;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.List;

public interface RoleService {
    List<Role> findAll();
    Page<Role> findAll(Pageable pageable);
    Page<Role> search(String query, Pageable pageable);
    Role findById(String id);
    Role create(Role role);
    Role update(String id, Role role);
    void delete(String id);
}

File: frontend/src/features/administration/roles/services/role.service.ts
import { apiClient } from "@/lib/axios";
import type { Page, PageRequest } from "@/types/common";
import type { CrudService } from "@/features/administration/shared/types/crud-service.types";
import { buildListParams } from "@/features/administration/shared/services/buildListParams";
import type { Role, RoleInput } from "../types/role.types";

export const roleService = {
  async list(params: PageRequest): Promise<Page<Role>> {
    const { data } = await apiClient.get<Page<Role>>("/roles", {
      params: buildListParams(params),
    });
    return data;
  },
  async create(input: RoleInput): Promise<Role> {
    const { data } = await apiClient.post<Role>("/roles", input);
    return data;
  },
  async update(id: string, input: RoleInput): Promise<Role> {
    const { data } = await apiClient.put<Role>(`/roles/${id}`, input);
    return data;
  },
  async remove(id: string): Promise<void> {
    await apiClient.delete(`/roles/${id}`);
  },
  async listAll(): Promise<Role[]> {
    const { data } = await apiClient.get<Role[]>("/roles/all");
    return data;
  },
} satisfies CrudService<Role, RoleInput>;
